From 2cf2c2785c164b7d56b7adeb2c3a9602bd9dc7bb Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sat, 1 Mar 2025 16:49:59 +0100
Subject: [PATCH] Scipy patch (#1)

* Update matdyn2fqha.py

* Update num.py

* Update signal.py

* Update thermo.py
---
 bin/matdyn2fqha.py | 4 ++--
 pwtools/num.py     | 5 +++--
 pwtools/signal.py  | 2 +-
 pwtools/thermo.py  | 5 +++--
 4 files changed, 9 insertions(+), 7 deletions(-)

diff --git a/bin/matdyn2fqha.py b/bin/matdyn2fqha.py
index d8d91c59..4823e9a2 100755
--- a/bin/matdyn2fqha.py
+++ b/bin/matdyn2fqha.py
@@ -34,14 +34,14 @@
 
 import sys
 import numpy as np
-from scipy.integrate import simps
+from scipy.integrate import simpson as simps
 
 filename = sys.argv[1]
 arr = np.loadtxt(filename)
 freq = arr[:,0]
 dos = arr[:,1]
 
-integral = simps(dos, freq)
+integral = simps(dos, x=freq)
 
 natom = integral / 3.
 nstep = len(freq)
diff --git a/pwtools/num.py b/pwtools/num.py
index cae61534..59a95492 100644
--- a/pwtools/num.py
+++ b/pwtools/num.py
@@ -6,7 +6,8 @@
 import scipy.optimize as optimize
 from scipy.interpolate import bisplrep, \
     bisplev, splev, splrep
-from scipy.integrate import simps, trapz
+from scipy.integrate import simpson as simps
+from scipy.integrate import cumulative_trapezoid as trapz
 from pwtools import _flib
 import warnings
 ##warnings.simplefilter('always')
@@ -104,7 +105,7 @@ def norm_int(y, x, area=1.0, scale=True, func=simps):
         fx = fy = 1.0
         sx, sy = x, y
     # Area under unscaled y(x).
-    _area = func(sy, sx) * fx * fy
+    _area = func(sy, x=sx) * fx * fy
     return y*area/_area
 
 
diff --git a/pwtools/signal.py b/pwtools/signal.py
index 3a3af11e..03948052 100644
--- a/pwtools/signal.py
+++ b/pwtools/signal.py
@@ -7,7 +7,7 @@
 import numpy as np
 from scipy.fftpack import fft, ifft
 from scipy.signal import fftconvolve, gaussian, kaiserord, firwin, lfilter, freqz
-from scipy.integrate import trapz
+from scipy.integrate import cumulative_trapezoid as trapz
 from pwtools import _flib, num
 
 
diff --git a/pwtools/thermo.py b/pwtools/thermo.py
index a3bdb13b..c5ff0316 100644
--- a/pwtools/thermo.py
+++ b/pwtools/thermo.py
@@ -1,7 +1,8 @@
 """(Quasi)harmonic approximation. Thermal expansion tools."""
 
 import numpy as np
-from scipy.integrate import simps, trapz
+from scipy.integrate import simpson as simps
+from scipy.integrate import cumulative_trapezoid as trapz
 from pwtools.constants import kb, hplanck, R, pi, c0, Ry_to_J, eV,\
     eV_by_Ang3_to_GPa
 from pwtools.verbose import verbose
@@ -151,7 +152,7 @@ def _norm_int(self, y, x, area):
         fy = np.abs(y).max()
         sx = x / fx
         sy = y / fy
-        _area = self.integrator(sy, sx) * fx * fy
+        _area = self.integrator(sy, x=sx) * fx * fy
         return y*area/_area
 
     def _printwarn(self, msg):
